services/provider/server/server.go

```diff
@@ -681,6 +681,17 @@ func (s *OCSProviderServer) getOCSSubscriptionChannel(ctx context.Context) (stri
 	if subscription == nil {
 		return "", fmt.Errorf("unable to find ocs-operator subscription")
 	}
+	if subscription.Status.InstalledCSV == "" {
+		return "", fmt.Errorf("subscription %s has no installedCSV", subscription.Name)
+	}
+	csv := &opv1a1.ClusterServiceVersion{}
+	err = s.client.Get(ctx, client.ObjectKey{Name: subscription.Status.InstalledCSV, Namespace: s.namespace}, csv)
+	if err != nil {
+		return "", fmt.Errorf("failed to get installed CSV %q: %w", subscription.Status.InstalledCSV, err)
+	}
+	if csv.Status.Phase != opv1a1.CSVPhaseSucceeded {
```

Comment:
a fair solution, I believe it'll be helpful not to block sending the channel if we are having issues w/ csv upgrade in patch versions.

rationale being, we aren't introducing any api change in patch versions and a failure here gets surfaced to client when the desiredstate is being processed.

Reply:
That will be tricky to detect.
As a customer might not always upgrade to 4.19.0.
They might for example upgrade from 4.18.4 to 4.19.2. And Sub doesn't store the previous CSV version for us to make a comparision.

Reply:
I mean we could just compare major & minor versions.